clippy_lints/src/utils/internal_lints/metadata_collector.rs

```diff
@@ -102,13 +127,33 @@ declare_clippy_lint! {
 impl_lint_pass!(MetadataCollector => [INTERNAL_METADATA_COLLECTOR]);
 
 #[allow(clippy::module_name_repetitions)]
-#[derive(Debug, Clone, Default)]
+#[derive(Debug, Clone)]
 pub struct MetadataCollector {
     /// All collected lints
     ///
     /// We use a Heap here to have the lints added in alphabetic order in the export
     lints: BinaryHeap<LintMetadata>,
     applicability_info: FxHashMap<String, ApplicabilityInfo>,
+    config: Vec<ClippyConfiguration>,
+}
+
+impl MetadataCollector {
+    pub fn new() -> Self {
```

Comment:
Doesn't this trigger our [``new_without_default``](https://rust-lang.github.io/rust-clippy/master/index.html#new_without_default) lint?

Reply:
It's the first time that I've heard of that lint... so, I guess not 🤔.  Unless it got merged after this branch was crated from master. I would also expect it not to trigger as `new()` does more than just assign default values but that might be just my opinion 🙃 .

It didn't trigger during testing. We'll see if bors complains about it 🙃